File: yggdrasil/examples/split_and_merge/src/split_and_merge_modelB.cpp
#include <iostream>
#include <math.h>

float modelB_function(float in_val) {
  float out_val = 3 * in_val;
  std::cout << "modelB_function(" << in_val << ") = " << out_val << std::endl;
  return out_val;
}

File: yggdrasil/examples/split_and_merge/src/split_and_merge_modelB.f90
function modelB_function(in_val, out_val) result(out)
  real(kind=8), intent(in) :: in_val
  real(kind=8), intent(out) :: out_val
  logical :: out
  out = .true.
  out_val = 3 * in_val
  write(*, '("modelB_function(",F10.5,") = (",F10.5,")")') in_val, out_val
end function modelB_function
